river/View.zig

```diff
@@ -476,6 +485,30 @@ pub fn applyConstraints(self: *Self, box: *wlr.Box) void {
     box.height = math.clamp(box.height, self.constraints.min_height, self.constraints.max_height);
 }
 
+/// Attach current view after n Views on the stack
+pub fn attach_aside(
+    view: *Self,
+    comptime state_field: @Type(.EnumLiteral),
+    comptime state_link_field: @Type(.EnumLiteral),
+    wm_stack: anytype,
+    tags: usize,
+    n: usize,
+) void {
```

Comment:
This level of metaprogramming should not be necessary. I just pushed a small refactor to master that should allow this function to only deal with the pending wm_stack.

Reply:
That factor does indeed help. Perhaps we can move the `tags` inside `pending` to keep similar structure as `Output.pending`?

Reply:
I'm not sure what `tags` you are referring to here, could you be more precise?

Reply:
I am referring to the two structs `Root.fallback.pending` and `Output.pending`. Seeing you have refactored `Root.fallback.inflight` away we can move `Root.fallback.tags` in `Root.fallback.pending` like this

**Root.zig:**
```zig
fallback: struct {
...
  pending: struct {
     tags: u32 = 1 << 0,
     focus_stack: wl.list.Head(View, .pending_focus_stack_link),
     wm_stack: wl.list.Head(View, .pending_wm_stack_link),
  },
...
},
```

That has the benefit of keeping the same struct as

**Output.zig:**
```zig
pending: struct {
    tags: u32 = 1 << 0,
    focus_stack: wl.list.Head(View, .pending_focus_stack_link),
    wm_stack: wl.list.Head(View, .pending_wm_stack_link),
},
```

while also simplifying the call to `attach_aside()` since we will only have to pass `Output.pending`/`Root.fallback.pending` depending on the situation.

Reply:
Ah, yeah making those the same type is a good idea. Just pushed 6e9bd83e1d0e2132d2def338b51f66867cd88907